euth/flashpoll/static/js/flashpoll.js

```diff
@@ -819,3 +826,117 @@ window.changeType = function(questionType, key, qorderId) {
     }
   }
 }
+
+function JSONToCSVConvertor(JSONData, ReportTitle) {
+    //If JSONData is not an object then JSON.parse will parse the JSON string in an Object
+    var arrData = typeof JSONData != 'object' ? JSON.parse(JSONData) : JSONData;
+    
+    var CSV = '';    
+    var row = "";
+    
+    // Fields to skip
+    var fieldsToSkip = ["userAgent", "userLocation"];
+    var timestampField = "timestamp";
+    
+    //This loop will extract the label from 1st index of on array
+    for (var index in arrData[0]) {
+        if(!contains(fieldsToSkip, index)){
+                //Now convert each value to string and comma-seprated
+                row += index + ',';
+        }        
+    }
+
+    row = row.slice(0, -1);
+    
+    //append Label row with line break
+    CSV += row + '\r\n';
+    
+    //1st loop is to extract each row
+    for (var i = 0; i < arrData.length; i++) {
+        var row = "";
+        
+        //2nd loop will extract each column and convert it in string comma-seprated
+        for (var index in arrData[i]) {
+            if(!contains(fieldsToSkip, index)){                
+                if(index == timestampField){
+                    row += '"' + (new Date(arrData[i][index]*1000)).toString() + '",';
+                }else{
+                    row += '"' + arrData[i][index] + '",';
+                }                
+            }
+        }
+
+        row.slice(0, row.length - 1);
+        
+        //add a line break after each row
+        CSV += row + '\r\n';
+    }
+
+    //Generate a file name
+    var fileName = "pollResults_";
+    //this will remove the blank-spaces from the title and replace it with an underscore
+    fileName += ReportTitle.replace(/ /g,"_");   
+    
+    //Initialize file format you want csv or xls
+    var uri = 'data:text/csv;charset=utf-8,' + escape(CSV);
+    
+    // Now the little tricky part.
+    // you can use either>> window.open(uri);
+    // but this will not work in some browsers
+    // or you will not get the correct file extension    
```

Comment:
Apparently, it is not possible to rename the file usint window.open().
The exported file name is always data.csv.
How can we handle this, I do want the file to  be named after the polI.